eks/cluster-loader/addon.go

```diff
@@ -0,0 +1,101 @@
+package clusterloader
+
+import (
+	"fmt"
+	"github.com/aws/aws-k8s-tester/eksconfig"
+	k8sclient "github.com/aws/aws-k8s-tester/pkg/k8s-client"
+	gotemplate "github.com/aws/aws-k8s-tester/pkg/util"
+	"io/ioutil"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+	"strings"
+)
+
+type ClusterLoader struct {
+	Config    *eksconfig.Config
+	K8sClient k8sclient.EKS
+}
+
+type ConfigMapFileStore struct {
+	Metadata v1.ObjectMeta
+	Data     map[string]string
+}
+
+// IsEnabled returns true if enabled
+func (c *ClusterLoader) IsEnabled() bool {
+	return c.Config.Spec.ClusterLoader != nil
+}
+
+// Apply installs the addon
+func (c *ClusterLoader) Apply() (err error) {
+	if c.Config.Spec.ClusterLoader.TestParams != nil {
+		for i, elem := range c.Config.Spec.ClusterLoader.TestParams {
+			if i < len(c.Config.Spec.ClusterLoader.TestParams)-1 {
+				c.Config.Spec.ClusterLoader.TestParams[i] = elem + ","
+			}
+		}
+	}
+	configmap := ConfigMapFileStore{
+		Metadata: v1.ObjectMeta{
+			Name: "cl2-config",
+		},
+		Data: c.formDataMap(),
+	}
+
+	template, err := gotemplate.FromLocalDirectory(struct {
+		*eksconfig.ClusterLoaderSpec
+		ConfigMap ConfigMapFileStore
+	}{
+		ClusterLoaderSpec: c.Config.Spec.ClusterLoader,
+		ConfigMap:         configmap,
+	})
+	c.K8sClient.Delete(template.String())
+	if err != nil {
+		return fmt.Errorf("while building templates, %v", err)
+	}
+	if err := c.K8sClient.Apply(template.String()); err != nil {
+		return fmt.Errorf("while applying resources, %v", err)
+	}
+	c.Config.Status.ClusterLoader = &eksconfig.ClusterLoaderStatus{
+		AddonStatus: eksconfig.AddonStatus{
+			Installed: true,
+			Ready:     true,
+		},
+	}
+	return nil
+}
+
+// Delete removes the addon
+func (c *ClusterLoader) Delete() (err error) {
+	template, err := gotemplate.FromLocalDirectory(c.Config.Spec.ClusterLoader)
+	if err != nil {
+		return fmt.Errorf("while building templates, %v", err)
+	}
+	if err := c.K8sClient.Delete(template.String()); err != nil {
+		return fmt.Errorf("while deleting resources, %v", err)
+	}
+	c.Config.Status.MetricsServer = &eksconfig.MetricsServerStatus{
+		AddonStatus: eksconfig.AddonStatus{
+			Installed: false,
+			Ready:     false,
+		},
+	}
+	return nil
+}
+
+func (c *ClusterLoader) formDataMap() map[string]string {
+	uris := c.Config.Spec.ClusterLoader.TestConfigUris
+	dataMap := make(map[string]string)
+	for _, uri := range uris {
+		basenameArr := strings.Split(uri, "/")
+		basename := basenameArr[len(basenameArr)-1]
+		resp, err := http.Get(uri)
+		if err != nil {
+			panic(err)
```

Comment:
Bad practice to panic. This will kill the process and completely stop AKT without error messages. 

You should return a decorated error all the way up the stack. Panic is reserved for situations where it would be dangerous to not stop immediately (typically in lower level libraries).
```
return fmt.Errorf("while downloading uri %s: %v, uri, err) 
```
This "while doing this" error pattern results in really nicely readable error messages.